src/Type/Php/ExplodeFunctionDynamicReturnTypeExtension.php

```diff
@@ -34,19 +39,55 @@ public function getTypeFromFunctionCall(
 		}
 
 		$delimiterType = $scope->getType($functionCall->args[0]->value);
-		$isSuperset = (new ConstantStringType(''))->isSuperTypeOf($delimiterType);
-		if ($isSuperset->yes()) {
+		$stringType = $scope->getType($functionCall->args[1]->value);
+
+		$nonEmptyStringArrayType = new IntersectionType([
+			new ArrayType(new IntegerType(), new StringType()),
+			new NonEmptyArrayType(),
+		]);
+
+		if ($delimiterType instanceof MixedType) {
+			return new BenevolentUnionType([
+				$nonEmptyStringArrayType,
+				new ConstantBooleanType(false),
+			]);
+		}
+
+		if (!$delimiterType instanceof ConstantStringType) {
```

Comment:
delimiterType could be a union of `ConstantStringType`s. Use `TypeUtils::getConstantStrings()` to find that out.